src/util/LockUtils.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Await this function to wait a set amount of time (without consuming cpu).
+ * @param delay - How long to wait.
+ * @param jitter - A fraction of this jitter will be added to the delay.
+ * @returns Promise<void> when ready.
+ */
+export async function wait(delay: number, jitter = 0): Promise<void> {
+  function calcTime(): number {
+    jitter = jitter > 0 ? Math.max(1, Math.floor(Math.random() * jitter)) : 0;
+    return Math.max(0, delay + jitter);
+  }
+  return new Promise<void>((resolve): any => setTimeout(resolve, calcTime()));
+}
+
+export interface AttemptSettings {
```

Comment:
It feels weird having this interface here, but not any function that references it. Looking at the `tryFn` of the new locker, it seems quite similar to the one from the redis locker. The only difference being what to do with the output of the function.

Perhaps this can be generalized in a function here as well, I was thinking of something like `function retryFunctionUntil<T>(fn: () => Promise<T>, settings: AttemptSettings): Promise<T>` which does all of the retry behaviour and then returns a promise with the resulting value or an error in case either the function throws an error or the timeout was reached.